services/actions/workflow.go

```
@@ -192,22 +192,48 @@ func DispatchActionWorkflow(ctx reqctx.RequestContext, doer *user_model.User, re
 
 	// find workflow from commit
 	var workflows []*jobparser.SingleWorkflow
-	for _, entry := range entries {
-		if entry.Name() != workflowID {
-			continue
-		}
+	var entry *git.TreeEntry
 
-		content, err := actions.GetContentFromEntry(entry)
-		if err != nil {
-			return err
-		}
-		workflows, err = jobparser.Parse(content)
-		if err != nil {
-			return err
+	for _, e := range entries {
+		if e.Name() != workflowID {
+			continue
 		}
+		entry = e
 		break
 	}
 
+	content, err := actions.GetContentFromEntry(entry)
```

Do we need to check if `entry` is nil before `GetContentFromEntry` ?

---

yep, fixing

---

I think we should return an error when `entry` is nil so that the user can know that the `workflowID` is incorrect.

```
if entry == nil {
	return util.ErrorWrapLocale(
		util.NewNotExistErrorf("workflow %q doesn't exist", workflowID),
		"actions.workflow.not_found", workflowID,
	)
}

content, err := actions.GetContentFromEntry(entry)
...
```